src/replayq.erl

```diff
@@ -307,59 +320,81 @@ transform(Id, [Item0 | Rest], Sizer, Count, Bytes, Acc) ->
 append_in_mem([], Q) -> Q;
 append_in_mem([Item | Rest], Q) -> append_in_mem(Rest, queue:in(Item, Q)).
 
-pop(Q, _Bytes, 0, AckRef, Acc) ->
+pop(Q, _Bytes, 0, AckRef, Acc, _StopFun, _StopFunAcc) ->
   Result = lists:reverse(Acc),
   ok = maybe_save_pending_acks(AckRef, Q, Result),
```

Comment:
unrelated, but this should probably called after each pop call (not only when full-batch is collected).

Reply:
I think we can make another ticket/PR to fix this.